RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitMomentumHelper.h

```diff
@@ -9,14 +9,14 @@ class VectorHitMomentumHelper {
   VectorHitMomentumHelper() {}
   ~VectorHitMomentumHelper() {}
 
-  float transverseMomentum(VectorHit& vh, const MagneticField* magField) {
+  float transverseMomentum(VectorHit& vh, const MagneticField* magField) const {
```

Comment:
change in namepace vectorHitMomentumHelper
and those methods in
constexpr

Reply:
I did implement this in the current code, but I am thinking that it might be easiest to just store the value of pT in the VectorHit directly. The only thing that is variable in the current calculation is the magnetic field strength, and I can't envision a scenario where one would want to recalculate the pT of the VectorHit with a different magnetic field. In any case, that would still be possible using the curvature that is also accessible if one really needs to. 

@mmusich @mtosi @vmariani am I overlooking something here?

Reply:
what about a change of the alignment ?
wouldn't it need to recompute the pT ? (because in principle the magnetic field cold be different)

Reply:
1) if alignment changes we reco again
2) magnetic field affects CPE more than pt
3) Using the local MF to compute pt is wrong anyhow

Reply:
Thanks for the input, I will implement it with the pT as a data member then.

Reply:
1. if alignment changes we reco again

what about refit?

Reply:
In the end I moved `momentum()` and `transverseMomentum()` back to the `VectorHit` class, but passing the magnetic field as a float and the object. That gets rid of the helper class and also allows the magnetic field to be updated.

Reply:
Awesome, thank you.